tests/integration/library.rs

```diff
@@ -43,9 +43,16 @@ version = "0.1.0"
         force_git_init: false,
         allow_commands: false,
     };
+
+    let app_config: AppConfig = app_config_path(&args_exposed.config)
+        .expect("cannot get app's config path")
+        .as_path()
+        .try_into()
+        .expect("cannot parse app's config path");
+
     // need to cd to the dir as we aren't running in the cargo shell.
     assert!(std::env::set_current_dir(&dir.root).is_ok());
-    assert!(generate(args_exposed).is_ok());
+    assert!(generate(app_config, args_exposed).is_ok());
```

Comment:
I would also look at the return value here - there's no test on the `Result<PathBuf>`, which was the point of this refactoring...

Reply:
I think the recently added test is fine, it verifies the returned path